java/dataset/src/main/cpp/jni_util.h

```diff
@@ -46,6 +47,8 @@ std::string JStringToCString(JNIEnv* env, jstring string);
 
 std::vector<std::string> ToStringVector(JNIEnv* env, jobjectArray& str_array);
 
+std::map<std::string, long> ToMap(JNIEnv* env, jobjectArray& str_array);
```

Comment:
This needs to be more descriptively named and needs a docstring because it has a very specific purpose.